src/app/chip/chip-input/chip-input.component.ts

```diff
@@ -0,0 +1,95 @@
+import { Component, ElementRef, EventEmitter, forwardRef, Input, NgZone, OnDestroy, OnInit, Output } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+
+@Component({
+  selector: 'mz-chip-input',
+  templateUrl: './chip-input.component.html',
+  styleUrls: ['./chip-input.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => MzChipInputComponent),
+      multi: true,
+    },
+  ],
+})
+export class MzChipInputComponent implements ControlValueAccessor, OnInit, OnDestroy {
+  @Input() autocompleteOptions: Materialize.AutoCompleteOptions;
+  @Input() placeholder: string;
+  @Input() secondaryPlaceholder: string;
+  @Output() add = new EventEmitter<Materialize.ChipDataObject>();
+  @Output() delete = new EventEmitter<Materialize.ChipDataObject>();
+  @Output() select = new EventEmitter<Materialize.ChipDataObject>();
+
+  get value(): Materialize.ChipDataObject[] {
+    return this.chipInputElement.material_chip('data') as Materialize.ChipDataObject[];
+  }
+
+  private chipInputElement: JQuery;
+  private onChangeCallback = (_: Materialize.ChipDataObject[]) => {};
```

Comment:
Can you explain me what this variable do?

Reply:
Its a callback function to propagate new value to `NgControl` which comes from the implementation of `NG_VALUE_ACCESSOR`